src/Microsoft.AspNetCore.Mvc.ApiExplorer/ApiBehaviorApiDescriptionProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ApiExplorer
+{
+    public class ApiBehaviorApiDescriptionProvider : IApiDescriptionProvider
+    {
+        private readonly IModelMetadataProvider _modelMetadaProvider;
+
+        public ApiBehaviorApiDescriptionProvider(IModelMetadataProvider modelMetadataProvider)
+        {
+            _modelMetadaProvider = modelMetadataProvider;
+        }
+
+        /// <remarks>
+        /// The order is set to execute after the default provider.
+        /// </remarks>
+        public int Order => -1000 + 10;
+
+        public void OnProvidersExecuted(ApiDescriptionProviderContext context)
+        {
+        }
+
+        public void OnProvidersExecuting(ApiDescriptionProviderContext context)
+        {
+            foreach (var description in context.Results)
+            {
+                if (!AppliesTo(description))
+                {
+                    continue;
+                }
+
+                foreach (var responseType in CreateProblemResponseTypes(description))
+                {
+                    description.SupportedResponseTypes.Add(responseType);
+                }
+            }
+        }
+
+        public bool AppliesTo(ApiDescription description)
+        {
+            return description.ActionDescriptor.FilterDescriptors.Any(f => f.Filter is IApiBehaviorMetadata);
+        }
+
+        public IEnumerable<ApiResponseType> CreateProblemResponseTypes(ApiDescription description)
+        {
+            if (description.ActionDescriptor.Parameters.Any() || description.ActionDescriptor.BoundProperties.Any())
+            {
+                // For validation errors.
+                yield return CreateProblemResponse(StatusCodes.Status400BadRequest);
```

Comment:
😢

Reply:
Is this always true? What about the `ApiBehaviorOptions`?

Reply:
This will be made extensible, but not in this PR 😆 

I'm just trying to remove a TODO that pranav added

Reply:
Just keeping you on your toes 😉

Reply:
It's good feedback, I'm adding an item here to track it. https://github.com/aspnet/Mvc/issues/6922